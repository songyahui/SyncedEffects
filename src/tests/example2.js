"use hopscript"

var hh = require("hiphop");

var prg = <hh.module T V>
  <hh.local S>
    <hh.loop>
      <hh.sequence>
	<hh.emit S/>
	<hh.if S>
	  <hh.emit T/>
	</hh.if>
	<hh.pause/>
	<hh.emit V/>
      </hh.sequence>
    </hh.loop>
  </hh.local>
</hh.module>

exports.prg = new hh.ReactiveMachine(prg, "example2");
