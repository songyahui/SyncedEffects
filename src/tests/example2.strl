module example2:

output T, V;

signal S in
       loop
               emit S;
	       present S then emit T end;
	       pause;
	       emit V;
       end loop
end signal

end module