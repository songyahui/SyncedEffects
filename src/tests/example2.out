example2> ;
--- Output: T
example2> ;
--- Output: T V
example2> ;
--- Output: T V
example2> ;
--- Output: T V
example2> ;
--- Output: T V
